plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/MultiSigStateMachine.hs

```diff
@@ -0,0 +1,159 @@
+cancelPayment
+    :: (WalletAPI m, WalletDiagnostics m)
+    => MultiSig.Params
+    -> MultiSig.State
+    -> m MultiSig.State
+cancelPayment prms st = mkStep prms st Cancel
+
+-- | Add a signature to a proposed payment
+addSignature
+    :: (WalletAPI m, WalletDiagnostics m)
+    => MultiSig.Params
+    -> MultiSig.State
+    -> m MultiSig.State
+addSignature prms st = WAPI.ownPubKey >>= mkStep prms st . AddSignature
+
+-- | Make a payment after enough signatures have been collected.
+makePayment
+    :: (WalletAPI m, WalletDiagnostics m)
+    => MultiSig.Params
+    -> MultiSig.State
+    -> m MultiSig.State
+makePayment prms st = do
+    -- we can't use 'mkStep' because the outputs of the transaction are 
+    -- different from the other transitions: We need two outputs, a public
+    -- key output with the payment, and the script output with the remaining
+    -- funds.
+    (currentValue, valuePaid, recipient) <- case st of
+        CollectingSignatures vl (Payment pd pk _) _ -> pure (vl, pd, pk)
+        _ -> WAPI.throwOtherError "Cannot make payment because no payment has been proposed. Run the 'proposePayment' action first."
+    
+    let newState = $$step prms st Pay
+        vl       = validator prms
+        redeemer = RedeemerScript (Ledger.lifted (SM.transition newState Pay))
+        dataScript = DataScript (Ledger.lifted (SM.transition newState Pay))
+
+    inputs <- WAPI.spendScriptOutputs (Ledger.scriptAddress vl) vl redeemer
+    let valueLeft = currentValue `Value.minus` valuePaid
+        scriptOut = Ledger.scriptTxOut valueLeft vl dataScript
+        pkOut     = Ledger.pubKeyTxOut valuePaid recipient
+    _ <- WAPI.createTxAndSubmit WAPI.defaultSlotRange (Set.fromList $ fmap fst inputs) [scriptOut, pkOut]
+    pure newState
+
+-- | Advance a running multisig contract. This applies the transition function 
+--   'SM.transition' to the current contract state and uses the result to unlock
+--   the funds currently in the contract, and lock them again with a data script
+--   containing the new state.
+--
+mkStep
+    :: (WalletAPI m, WalletDiagnostics m)
+    => MultiSig.Params
+    -- ^ The parameters of the contract instance
+    -> MultiSig.State
+    -- ^ Current state of the instance
+    -> MultiSig.Input
+    -- ^ Input to be applied to the contract
+    -> m MultiSig.State
+    -- ^ New state after applying the input
+mkStep prms st input = do
+    let newState = $$step prms st input
+        vl       = validator prms
+        redeemer = RedeemerScript (Ledger.lifted (SM.transition newState input))
+        dataScript = DataScript (Ledger.lifted (SM.transition newState input))
+
+    inputs <- WAPI.spendScriptOutputs (Ledger.scriptAddress vl) vl redeemer
+    let totalVal = foldMap snd inputs
+        output = Ledger.scriptTxOut totalVal vl dataScript
+    _ <- WAPI.createTxAndSubmit WAPI.defaultSlotRange (Set.fromList $ fmap fst inputs) [output]
+    pure newState
+
+{- note [Current state of the contract]
```

Comment:
I think it's common to start notes with `Note` rather than `note` and it's convenient for grepping to use the same notation.